src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -56,3 +59,34 @@ def get(self, svc_id):
             'osd_metadata': mgr.get_metadata('osd', svc_id),
             'histogram': histogram,
         }
+
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def flags(self):
+        enabled_flags = mgr.get('osd_map')['flags_set']
+        if 'pauserd' in enabled_flags and 'pausewr' in enabled_flags:
+            # 'pause' is set by calling `ceph osd set pause` and unset by
+            # calling `set osd unset pause`, but `ceph osd dump | jq '.flags'`
+            # will contain 'pauserd,pausewr' if pause is set.
+            # Let's pretend to the API that 'pause' is in fact a proper flag.
+            enabled_flags = list(
+                set(enabled_flags) - {'pauserd', 'pausewr'}) + ['pause']
+        if 'Content-Length' in cherrypy.request.headers != '':
+            content_length = int(cherrypy.request.headers['Content-Length'])
+            body = cherrypy.request.body.read(content_length)
```

Comment:
why can't we receive the POST body data through method parameters? 

Couldn't you declare the method as: `def flags(self, **kwargs)` ?